File: Arrays/167. Two Sum II - Input Array Is Sorted.cpp
// Url - https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right = numbers.size() - 1;
        vector<int> ans;

        int tempSum = 0;
        while(left < right)
        {
            tempSum = numbers[left] + numbers[right];
            if(tempSum == target)
            {
                ans.push_back(left + 1);
                ans.push_back(right + 1);
                left++;
                right--;
            }
            else if(tempSum < target)
            {
                left++;
            }
            else
            {
                right--;
            }
        }
        
        return ans;
    }
};

File: Arrays/167. Two Sum II - Input Array Is Sorted.py
# Url - https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        
        ans = []
        
        while left < right:
            if (numbers[left] + numbers[right]) == target:
                ans.append(left + 1)
                ans.append(right + 1)
                left += 1 
                right -= 1
            elif (numbers[left] + numbers[right]) < target:
                left += 1
            else:
                right -= 1
                
        return ans
                